core/include/detray/intersection/cylinder_intersector.hpp

```diff
@@ -0,0 +1,185 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/definitions/qualifiers.hpp"
+#include "detray/intersection/detail/trajectories.hpp"
+#include "detray/intersection/intersection.hpp"
+#include "detray/utils/quadratic_equation.hpp"
+
+// System include(s)
+#include <cmath>
+#include <type_traits>
+
+namespace detray {
+
+/** A functor to find intersections between trajectory and cylinder mask
+ */
+struct cylinder_intersector {
```

Comment:
OK.

Reply:
But let me defend myself one more time

>  The way the cylinder is handled in the navigator after your changes, for example, would not work with a helix trajectory 

Why would this happen? Shouldn't there be only one difference in `path_length` of intersection objects?

Reply:
>  specialized intersectors for vectorization (I am not sure that is necessary, yet),

Let's not do that unless there is a good reason for that

Reply:
> > specialized intersectors for vectorization (I am not sure that is necessary, yet),
> 
> Let's not do that unless there is a good reason for that

I agree, it might not change much after we deploy the local navigation (currently it is the largest hotspot though), but we might use (even ray) intersections in accelerator structures. The reason we will need different intersectors outside algebra-plugins is, that the way boolean checks need to be handled is different, as well as e.g. the order or type of algebra operations. The intersectors already exist, they are just not deployed in detray, yet. Besides, that is a sort of 'easy' use case to try and see what it takes to go to SoA layout or whether to abandon that...